src/webgpu/idl/javascript.spec.ts

```diff
@@ -356,6 +369,38 @@ g.test('limits')
     }
   });
 
+g.test('readonly_properties')
+  .desc(
+    `
+    Test that setting a property with no setter throws.
+    `
+  )
+  .params(u => u.combine('type', kResources))
+  .fn(t => {
```

Comment:
nit: Our build settings enable `'use strict'`, but best to do it here explicitly, just in case someone were to build it with different options.

If there is a compiler warning or something (since the compiler knows this won't actually do anything in our build), you can suppress it if possible, otherwise never mind.

```suggestion
  .fn(t => {
    'use strict'; // This makes setting a readonly property produce a TypeError.
```

Reply:
added